### DISCUSSION TITLE
Okta Authentication Returns 404 From Netbox

### DISCUSSION THREAD

**Discussion by kyle-burnett** [netbox-community/netbox#10490] [Q&A] [answered]
### Netbox Version
v3,3,4

### Python Version
Python 3.10.4

I feel like I'm missing something incredibly stupid here.

I'm setting up Okta authentication as per https://docs.netbox.dev/en/stable/administration/authentication/okta/. On the login page I'm able to see the Okta authentication button which links to `https://netbox.example.com/oauth/login/okta-openidconnect/`

Clicking the Okta auth button immediately redirects me to `https://netbox.example.com/oauth/login/okta-openidconnect/v1/authorize` and I get a 404 result
```
Request Method:	GET
Request URL: https://netbox.example.com/oauth/login/okta-openidconnect/v1/authorize?client_id=xxxxxxxxx&redirect_uri=https://netbox.example.com/oauth/complete/okta-openidconnect/&state=xxxxxxx&response_type=code&nonce=xxxxxxxx&scope=openid+profile+email
Using the URLconf defined in netbox.urls, Django tried these URL patterns, in this order:

[name='home']
search/ [name='search']
login/ [name='login']
logout/ [name='logout']
oauth/ login/<str:backend>/ [name='begin']
oauth/ complete/<str:backend>/ [name='complete']
oauth/ disconnect/<str:backend>/ [name='disconnect']
oauth/ disconnect/<str:backend>/<int:association_id>/ [name='disconnect_individual']
circuits/
dcim/
extras/
ipam/
tenancy/
user/
virtualization/
wireless/
api/ [name='api-root']
api/circuits/
api/dcim/
api/extras/
api/ipam/
api/tenancy/
api/users/
api/virtualization/
api/wireless/
api/status/ [name='api-status']
api/docs/ [name='api_docs']
api/redoc/ [name='api_redocs']
^api/swagger(?P<format>.json|.yaml)$ [name='schema_swagger']
graphql/ [name='graphql']
media/<path:path>
media-failure/ [name='media_failure']
plugins/
api/plugins/
admin/background-tasks/
admin/plugins/
admin/
__debug__/
metrics [name='prometheus-django-metrics']

The current path, oauth/login/okta-openidconnect/v1/authorize, didn’t match any of these.
```

I don't see it ever try and connect to the Okta domain defined under `SOCIAL_AUTH_OKTA_OPENIDCONNECT_API_URL`.

**Comment by kyle-burnett** [comment, marked as answer]
> I feel like I'm missing something incredibly stupid here.

Yup...I'm running Netbox-Docker in a k8s cluster, and I guess `SOCIAL_AUTH_OKTA_OPENIDCONNECT_API_URL`, `SOCIAL_AUTH_OKTA_OPENIDCONNECT_KEY`, & `SOCIAL_AUTH_OKTA_OPENIDCONNECT_SECRET` aren't loaded into configuration.py with the rest of the environment variables.

I loaded them into a [custom configuration file](https://github.com/netbox-community/netbox-docker/wiki/Configuration#custom-configuration-files) and it seems to be working better now.

**Comment by ahmedabrar914** [comment]
Hi I need your help in setting up okta with Netbox, I have followed the https://netboxlabs.com/docs/netbox/en/stable/administration/authentication/okta/ to provision with my netbox, But I am getting  the error which is shown in image
<img width="1067" alt="Screenshot 2024-08-20 at 9 01 41 PM" src="https://github.com/user-attachments/assets/027aab74-1033-4eb8-9c9b-2fb49bfa3816">
